File: java/src/main/java/mchehab/com/java/PersonViewHolder.java
package mchehab.com.java;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

public class PersonViewHolder extends RecyclerView.ViewHolder{

    private View view;
    private TextView textViewFirstName;
    private TextView textViewLastName;

    public PersonViewHolder(View view) {
        super(view);
        this.view = view;
        textViewFirstName = view.findViewById(R.id.textViewFirstName);
        textViewLastName = view.findViewById(R.id.textViewLastName);
    }

    public View getView(){
        return  view;
    }

    public void bindData(Person person){
        textViewFirstName.setText(person.getFirstName());
        textViewLastName.setText(person.getLastName());
    }
}

File: app/src/main/java/mchehab/com/advancedrecyclerview/PersonViewHolder.kt
package mchehab.com.advancedrecyclerview

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView


class PersonViewHolder(val view: View): RecyclerView.ViewHolder(view) {

    private val textViewFirstName = view.findViewById<TextView>(R.id.textViewFirstName)
    private val textViewLastName = view.findViewById<TextView>(R.id.textViewLastName)

    fun bindData(person: Person){
        textViewFirstName.text = person.firstName
        textViewLastName.text = person.lastName
    }
}